docs/CHANGES.md

```diff
@@ -0,0 +1,228 @@
+    submit_withdrawals: SubmitWithdrawals,
+    submit_transactions: SubmitTransactions,
+}
+```
+
+* Pay fee with Simple UDT is removed in the v1, `registry_id` is added to `Fee` to indicate which registry the token in transfered to.
+
+``` molecule
+struct Fee {
+    // registry id
+    registry_id: Uint32,
+    // amount in CKB
+    amount: Uint64,
+}
+```
+
+* Ethereum address registry data structures.
+
+``` molecule
+// --- ETH Address Registry ---
+union ETHAddrRegArgs {
+    EthToGw,
+    GwToEth,
+    SetMapping,
+    BatchSetMapping,
+}
+
+struct EthToGw {
+    eth_address: Byte20,
+}
+
+struct GwToEth {
+    gw_script_hash: Byte32,
+}
+
+struct SetMapping {
+    gw_script_hash: Byte32,
+    fee: Fee,
+}
+
+table BatchSetMapping {
+    gw_script_hashes: Byte32Vec,
+    fee: Fee,
+}
+
+// --- end of ETH Address Registry ---
+```
+
+### Address registry
+
+1. A new builtin contract Ethereum address registry is added, Godwoken using it to handling the Ethereum address. When user deposit token to create a new account, a corresponded Ethereum address of the new account is inserted to the contract.
+2. Deposit automatically mapping the Ethereum address for new accounts. If the account is created through Meta contract, the developer must register the Ethereum address for the acount by calling the Ethereum address registry contract.
+3. The builtin Ethereum address registry is allocated to id 2 in the Godwoken genesis block.
+
+### Godwoken syscalls
+
+1. Remove `sys_get_script_hash_by_prefix_fn`
+2. Update the `sys_pay_fee`, the second param is registry address.
+3. Add `sys_get_registry_address_by_script_hash`
+4. Add `sys_get_script_hash_by_registry_address`
+
+``` c
+/**
+ * Record fee payment
+ *
+ * @param payer_addr                  Registry address
+ * @param sudt_id                     Account id of sUDT
+ * @param amount                      The amount of fee
+ * @return                            The status code, 0 is success
+ */
+typedef int (*gw_pay_fee_fn)(struct gw_context_t *ctx, gw_reg_addr_t payer_addr,
+                             uint32_t sudt_id, uint128_t amount);
+
+/**
+ * Get registry address by script_hash
+ *
+ * @param script_hash
+ * @param reg_id registry_id
+ * @param returned registry address
+ * @return       The status code, 0 is success
+ */
+typedef int (*gw_get_registry_address_by_script_hash_fn)(
+    struct gw_context_t *ctx, uint8_t script_hash[32], uint32_t reg_id,
+    gw_reg_addr_t *address);
+
+/**
+ * Get script hash by address
+ *
+ * @param address
+ * @param script_hash
+ * @return       The status code, 0 is success
+ */
+typedef int (*gw_get_script_hash_by_registry_address_fn)(
+    struct gw_context_t *ctx, gw_reg_addr_t *address, uint8_t script_hash[32]);
+```
+
+### Signing
+
+In v0 we only support sign transactions with personal sign, this is considered insecure since users can only see a random 32-bytes hex, they have no idea what data they are signing.
```

Comment:
```suggestion
In v0 we only support signing transactions with personal sign, this is considered insecure since users can only see a random 32-bytes hex, they have no idea what data they are signing.
```